plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentsRestControllerIntegrationTest.php

```diff
@@ -420,6 +529,218 @@ public function test_get_payment_providers_with_invalid_location() {
 		$this->assertEquals( 400, $response->get_status() );
 	}
 
+	/**
+	 * Test getting payment providers returns all the data.
+	 */
+	public function test_get_payment_providers_has_all_the_data() {
+		// Arrange.
+		// phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.Found
+		$filter_callback = fn( $caps ) => array(
+			'manage_woocommerce' => true,
+			'install_plugins'    => true,
+		);
+		add_filter( 'user_has_cap', $filter_callback );
+
+		$this->enable_core_paypal_pg();
+		$this->mock_payment_gateways();
+
+		// Act.
+		$request = new WP_REST_Request( 'GET', self::ENDPOINT . '/providers' );
+		$request->set_param( 'location', 'US' );
+		$response = $this->server->dispatch( $request );
+
+		// Assert.
+		$this->assertEquals( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		// We have the core PayPal gateway and the fake WooPayments gateway registered, the offline PMs group entry, and 2 suggestions.
+		$this->assertCount( 5, $data['providers'] );
+		// We also have the 3 offline payment methods.
+		$this->assertCount( 3, $data['offline_payment_methods'] );
+		// We get all the suggestions.
+		$this->assertCount( 7, $data['suggestions'] );
+		// Assert we get the suggestion categories.
+		$this->assertCount( 3, $data['suggestion_categories'] );
+
+		// Assert that the preferred suggestions are WooPayments and PayPal (full stack).
+		// The order is different because of the presence of the fake WooPayments gateway.
+		$this->assertEquals( Payments::SUGGESTION_ORDERING_PREFIX . PaymentExtensionSuggestions::WOOPAYMENTS, $data['providers'][0]['id'] );
+		$this->assertEquals( Payments::SUGGESTION_ORDERING_PREFIX . PaymentExtensionSuggestions::PAYPAL_FULL_STACK, $data['providers'][2]['id'] );
+
+		// Assert that the WooPayments suggestion has all the details, including the incentive data.
```

Comment:
What is the benefit of asserting _all_ the fields? Is it a type safety measure?

Reply:
I consider this a backstop against changes that may unwillingly alter the returned data. It is slightly paranoid, but I want us to know that our changes modified the API response.